estimagic/dashboard/convergence_plot.py

```diff
@@ -0,0 +1,157 @@
+        new_params, fitness = queue.get()
+
+        doc.add_next_tick_callback(
+            partial(_update_convergence_plot, data=param_data, new_values=new_params)
+        )
+        doc.add_next_tick_callback(
+            partial(_update_convergence_plot, data=fitness_data, new_values=fitness)
+        )
+
+
+def _fitness_plot(initial_fitness):
+    fitness_data = ColumnDataSource(
+        data=_convert_sr_for_cds(sr=initial_fitness, iteration=0)
+    )
+    fitness_p = _wide_figure(title="Fitness")
+    fitness_p.line(
+        source=fitness_data,
+        x="iteration",
+        y="fitness",
+        line_width=1,
+        name="fitness",
+        color="firebrick",
+    )
+
+    return fitness_p, fitness_data
+
+
+def _parameter_plots(param_df):
+    """
+    Create the plots that show the convergence of (groups of) parameters.
+
+    ToDo: split up convergence plot depending on column in param_df.
+    ToDo: plot upper and lower bounds.
+    ToDo: only plot parameters that are not fixed.
+
+    Args:
+        param_df (pandas DataFrame):
+            DataFrame with the initial parameter values, constraints etc.
+
+
+    """
+
+    assert "iteration" not in param_df.index, (
+        "Estimagic uses the key 'iteration'. "
+        + "Therefore, it may not be used in the index of the parameter DataFrame."
+    )
+
+    colors = _choose_color_palettes(param_df)
+
+    first_entry = _convert_sr_for_cds(sr=param_df["value"], iteration=0)
+    param_data = ColumnDataSource(data=first_entry)
+
+    conv_p = _wide_figure(title="Parameter Values")
+
+    _add_convergence_lines(figure=conv_p, param_data=param_data, colors=colors)
+
+    return [conv_p], [param_data]
+
+
+def _choose_color_palettes(param_df):
+    # color tone palettes: bokeh.palettes.Blues9, Greens9, Reds9, Purples9.
+    long_colors = bokeh.palettes.Viridis256
+    return random.sample(long_colors, len(param_df))
+
+
+def _add_convergence_lines(figure, param_data, colors):
+    line_names = [str(x) for x in sorted(param_data.column_names) if x != "iteration"]
+    for i, name in enumerate(line_names):
+        figure.line(
+            source=param_data,
+            x="iteration",
+            y=name,
+            line_width=1,
+            name=name,
+            color=colors[i],
+            nonselection_alpha=0,
+        )
+
+
+def _convert_sr_for_cds(sr, iteration):
+    """
+    Convert parameter Series for adding it to a ColumnDataSource.
+
+    Args:
+        sr (pd.Series):
+            Series with the new value(s)
+
+        iteration (int):
+            iteration of the parameter vector
+    """
+    entry = {"iteration": [iteration]}
+    entry.update({str(k): [v] for k, v in sr.to_dict().items()})
+    return entry
+
+
+@gen.coroutine
+def _update_convergence_plot(new_values, data):
+    iteration = max(data.data["iteration"]) + 1
+    to_add = _convert_sr_for_cds(sr=new_values, iteration=iteration)
+    data.stream(to_add)
+
+
+def _wide_figure(title):
```

Comment:
Is it robust to hardcode the size? How will it look on a beamer? We could try this by using a low resolution on a laptop.